prov/util/src/util_cntr.c

```diff
@@ -350,3 +432,85 @@ ofi_ep_cntr_inc_func ofi_ep_rx_cntr_inc_funcs[] = {
 	[ofi_op_atomic_compare] = ofi_ep_rem_rd_cntr_inc,
 	[ofi_op_read_async] = ofi_ep_rem_rd_cntr_inc,
 };
+
+static
+struct util_cntr *ofi_ep_get_tx_cntr_from_op(struct util_ep *ep, uint8_t op)
```

Comment:
This switch may not be optimal, but I am hesitated to do the same thing for the current util cntr implementation that uses an array for function pointers, which finally creates a bunch of functions for different cntrs. 

I am open to suggestions here.

Reply:
I see, thanks. Does that mean you would like to change the current `util_ep`

```
struct util_ep {
	...
	/* CNTR entries */
	struct util_cntr	*tx_cntr;     /* transmit/send */
	struct util_cntr	*rx_cntr;     /* receive       */
	struct util_cntr	*rd_cntr;     /* read          */
	struct util_cntr	*wr_cntr;     /* write         */
	struct util_cntr	*rem_rd_cntr; /* remote read   */
	struct util_cntr	*rem_wr_cntr; /* remote write  */

	ofi_cntr_inc_func	tx_cntr_inc;
	ofi_cntr_inc_func	rx_cntr_inc;
	ofi_cntr_inc_func	rd_cntr_inc;
	ofi_cntr_inc_func	wr_cntr_inc;
	ofi_cntr_inc_func	rem_rd_cntr_inc;
	ofi_cntr_inc_func	rem_wr_cntr_inc;
...
```

to something like 

struct util_ep {
    ...
    struct util_cntr *cntrs[CNTR_CNT];
    ofi_cntr_inc_func inc_functions[CNTR_CNT];
    ...
};
```
?

Reply:
I'd be good with that change.  If it helps with the shm and peer implementation, I'd say go ahead.  But otherwise I wouldn't consider it critical.

Reply:
Thanks, I will go that direction. It will make the code cleaner.

Reply:
I made this refactor as a separate PR as it changes a lot of providers: https://github.com/ofiwg/libfabric/pull/9127. I will rebase this PR on top of it after its merged.

Reply:
rebased